ios/authentication.md

```diff
@@ -1,344 +1,417 @@
+- Customize email and SMS messages for Multi-Factor Authentication
+- Customize attributes for your users, e.g. name, email
+- Enable 3rd party authentication providers, e.g. Facebook, Twitter, Google and Amazon
 
-2. Choose to set up your own configuration.
+After configuring your Authentication options, update your backend:
 
-    ```
-    ❯ No, I will set up my own configuration.
-    ```
+```bash
+$ amplify push
+```
 
-3. Choose to set up authentication flow using AWS IAM access controls.
+A configuration file called `awsconfiguration.json` will be copied to your project source directory. In the Finder, drag the file into Xcode under the top Project Navigator folder. When the `Options` dialog box that appears, do the following:
 
-    ```
-    ❯ User Sign-Up, Sign-In, connected with AWS IAM controls
-    ```
+* Clear the `Copy items if needed` check box.
+* Choose `Create groups`, and then choose `Next`.
 
-4. Choose yes, to: `? Allow unauthenticated logins?`.
+[Click here to learn more about this process.](./start#step-3-how-it-works)
 
-5. Choose yes, to: `? Do you want to enable 3rd party authentication providers in your identity pool?`.
+## Manual Setup
 
-6. Choose Facebook and then provide your Facebook app ID. To retrieve or create your Facebook app ID, see [Setting Up Facebook Authentication](http://docs.aws.amazon.com/aws-mobile/latest/developerguide/auth-facebook-setup.html).
+For manual configuration without the CLI, you must have an `awsconfiguration.json` file with the following:
+- Cognito User Pools: `CognitoUserPool : { Default: ...}`
+- Cognito Identity Pools: `IdentityManager` and `CredentialsProvider: {CognitoIdentity: ...}`
 
-7. When configuration for Facebook sign-in is complete, the CLI displays a message confirming that you have configured local CLI metadata for this category. You can confirm this by viewing status.
+```xml
+    {
+        "IdentityManager": {
+            "Default": {}
+        },
+        "CredentialsProvider": {
+            "CognitoIdentity": {
+                "Default": {
+                    "PoolId": "XX-XXXX-X:XXXXXXXX-XXXX-1234-abcd-1234567890ab",
+                    "Region": "XX-XXXX-X"
+                }
+            }
+        },
+        "CognitoUserPool": {
+            "Default": {
+                "PoolId": "XX-XXXX-X_abcd1234",
+                "AppClientId": "XXXXXXXX",
+                "AppClientSecret": "XXXXXXXXX",
+                "Region": "XX-XXXX-X"
+            }
+        }
+    }
+```
 
-    ```
-    $ amplify status
-    | Category  | Resource name   | Operation | Provider plugin   |
-    | --------- | --------------- | --------- | ----------------- |
-    | Auth      | cognitoa7cbb553 | Create    | awscloudformation |
-    ```
+If you are using both Cognito User Pools and Identity Pools, such as in Federated scenarios, you will need all of the keys mentioned above.
 
-8. To create your backend AWS resources run the following:
+## Initialization
```

Comment:
I think we have missed the `Podfile` entries step before this step.

Reply:
Good catch, added